src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs

```diff
@@ -1203,7 +1203,8 @@ static string GetBlobName(TableEntity entity, string property)
             }
             catch (DurableTaskStorageException ex)
             {
-                if (ex.HttpStatusCode == (int)HttpStatusCode.PreconditionFailed)
+                // The status code is "Conflict" for a null eTagValue and TableTransactionActionType.Add, and "PreconditionFailed" for a non-null eTagValue and TableTransactionActionType.UpdateMerge
+                if (ex.HttpStatusCode == (int)HttpStatusCode.Conflict || ex.HttpStatusCode == (int)HttpStatusCode.PreconditionFailed)
```

Comment:
ultra nit -

I think it's useful to structure htis in terms of what's the 'general case' versus the edge case.
In general, what you're detecting here is that the ETag has changed (PreconditionFailed).
And the edge case is that if the ETag changed _from `null`_ to a value, thne the status if 'Conflict'.

So I recommend organizing this in the order of 'most expected' to 'edge case', to make it easier to parse.
Like this:

```suggestion
                // Handle case where the 'ETag' has changed. This implies another worker has taken over this
                // work item while we were trying to process it. We detect this in 2 cases:
                // Common case: the resulting code is 'PreconditionFailed', it means our ETag is no longer current.
                // Edge case: the resulting code is 'Conflict'. This can occur if this was the first orchestrator work item, 
                // as no ETag exists then. The 'Conflict' represents that we attempted to create a history that already
                // existed .
                if (ex.HttpStatusCode == (int)HttpStatusCode.Conflict || ex.HttpStatusCode == (int)HttpStatusCode.PreconditionFailed)
```

Reply:
@sophiatev - in the 'conflict' case - should we not also be checking if ETag is 'null'? Otherwise, we're assuming it is null. Not sure, you're the expert :)

Reply:
Good idea :)